pkg/controller/controllercmd/builder.go

```diff
@@ -328,8 +348,19 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
 	// ensure blocking TCP connections don't block the leader election
 	leaderConfig := rest.CopyConfig(protoConfig)
 	leaderConfig.Timeout = b.leaderElection.RenewDeadline.Duration
+	var leaderElection leaderelection.LeaderElectionConfig
+
+	switch b.leaderElectionResourceLock {
+	// default to configmapsleases for leader election
+	// TODO: In the next version we should switch to using "leases" by default
+	case "", ConfigMapsLeasesResourceLock:
+		leaderElection, err = leaderelectionconverter.ToLeaderElectionWithConfigmapLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
+	case LeasesResourceLock:
+		leaderElection, err = leaderelectionconverter.ToLeaderElectionWithLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
+	default:
+		return fmt.Errorf("%w: %s", ErrInvalidResourceLock, b.leaderElectionResourceLock)
+	}
 
-	leaderElection, err := leaderelectionconverter.ToLeaderElectionWithConfigmapLease(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity)
```

Comment:
use proposed `eaderelectionconverter.ToLeaderElection(leaderConfig, *b.leaderElection, b.componentName, b.instanceIdentity, b.leaderElectionResourceLock`

we can get rid of the switch here